pro/src/commons/utils/__specs__/memoize.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { memoize } from "../memoize";
+
+describe('memoize', () => {
+  it('should return the correct result for the first call', () => {
+    const add = (a: number, b: number) => a + b;
+    const memoizedAdd = memoize(add);
+
+    expect(memoizedAdd(2, 3)).toBe(5);  // Expected result: 5
+  });
+
+  it('should return the cached result for subsequent calls with the same arguments', () => {
+    const add = (a: number, b: number) => a + b;
+    const memoizedAdd = memoize(add);
+
+    // First call - computes the result and stores it in cache
+    const result1 = memoizedAdd(2, 3);
+    expect(result1).toBe(5); 
+
+    // Second call with same arguments - uses cached result
+    const result2 = memoizedAdd(2, 3);
+    expect(result2).toBe(5);
+
+    // Ensure the function was only called once
+    expect(result1).toBe(result2);  // Should be the same value
```

Comment:
Ce cas de test ne couvre pas le comportement d'un véritable memoize.
Il faudrait pour s'en assurer vraiment vérifier qu'au second appel la fonction `add()` n'ait pas été invoquée.

```suggestion
    const _callee = vi.fn();
    
    const add = (a: number, b: number) => {
      _callee();
      return a + b;
    }
    const memoizedAdd = memoize(add);

    // First call - computes the result and stores it in cache
    const result1 = memoizedAdd(2, 3);
    expect(result1).toBe(5);
    expect(_callee).toHaveBeenCalledOnce()

    // Second call with same arguments - uses cached result
    const result2 = memoizedAdd(2, 3);
    expect(result2).toBe(5);
    expect(_callee).toHaveBeenCalledOnce()

    // Ensure the function was only called once
    expect(result1).toBe(result2);  // Should be the same value
```